catalog-info.yaml

```
@@ -45,6 +45,44 @@ spec:
         filter_enabled: true
         filter_condition: |
           build.pull_request.id == null || (build.creator.name == 'elasticmachine' && build.pull_request.id != null)
+      teams:
+        ingest-fp:
+          access_level: MANAGE_BUILD_AND_READ
+        everyone:
+          access_level: READ_ONLY
+
+---
+# yaml-language-server: $schema=https://gist.githubusercontent.com/elasticmachine/988b80dae436cafea07d9a4a460a011d/raw/e57ee3bed7a6f73077a3f55a38e76e40ec87a7cf/rre.schema.json
+apiVersion: backstage.io/v1alpha1
+kind: Resource
+metadata:
+  name: buildkite-pipeline-fleet-server-package-mbp
+  description: Buildkite pipeline for the Elastic Fleet Server package process
+  links:
+    - title: Pipeline
+      url: https://buildkite.com/elastic/fleet-server-package-mbp
+
+spec:
+  type: buildkite-pipeline
+  owner: group:ingest-fp
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: fleet-server-package-mbp
+      description: Buildkite pipeline for the Elastic Fleet Server package process
+    spec:
+      repository: elastic/fleet-server
+      pipeline_file: ".buildkite/pipeline.package.mbp.yml"
+      branch_configuration: "main 8.* 7.* v8.* v7.*"
+      provider_settings:
+        build_pull_request_forks: false
+        build_pull_requests: true
+        build_tags: false
+        filter_enabled: true
+        filter_condition: |
+          build.pull_request.id == null || (build.creator.name == 'elasticmachine' && build.pull_request.id != null)
```

Looking where this pipeline is executed. It looks like it just runs when it is triggered from the main pipeline.

https://github.com/elastic/fleet-server/commits/main
https://github.com/elastic/fleet-server/commits/8.8

Did you see any other location where it is triggered from some push event @sharbuz ?

If it does not, I would change the configuration to disallow `build_branches`

```suggestion
      provider_settings:
        build_pull_request_forks: false
        build_pull_requests: true
        build_tags: false
        build_branches: false
        filter_enabled: true
        filter_condition: |
          build.pull_request.id == null || (build.creator.name == 'elasticmachine' && build.pull_request.id != null)
```

---

Not sure how to allow triggering this to debug.
One option is to add a new entry in the buildkite-pr-bot configuration `.buildkite/pull-requests.json` file. 

That new entry should follow the same settings as this one

---

I didn't see any places where it executes with the exception of the main pipeline.

---

As this step is intended to be run just when the main pipeline triggers it (in some specific branches), I think it could be added the buildkite-pr-bot configuration.

That would help you to debug/develop this Buildkite pipeline.
You can follow the same settings as in https://github.com/elastic/elastic-agent/blob/main/.buildkite/pull-requests.json#L19-L34